packages/vscode-extension/src/debug/commonUtils.ts

```diff
@@ -109,36 +109,75 @@ export async function getDebugConfig(
   env?: string
 ): Promise<{ appId: string; env?: string } | undefined> {
   try {
-    if (isLocalSideloadingConfiguration) {
-      const localEnvManager = new LocalEnvManager(VsCodeLogInstance, ExtTelemetry.reporter);
-      const localSettings = await localEnvManager.getLocalSettings(ext.workspaceUri.fsPath);
-      return { appId: localSettings?.teamsApp?.teamsAppId as string };
+    if (isV3()) {
+      const inputs = getSystemInputs();
+      const getConfigRes = await core.getProjectConfigV3(inputs);
+      if (getConfigRes.isErr()) throw getConfigRes.error;
+      const config = getConfigRes.value;
+      if (!config)
+        throw new UserError("GetConfigError", "Failed to get project config", "Extension");
+      if (isLocalSideloadingConfiguration) {
+        const envInfo = config.envInfos["local"];
+        if (!envInfo)
+          throw new UserError("EnvConfigNotExist", "Local Env config not exist", "Extension");
+        const appId = envInfo.state["fx-resource-appstudio"].teamsAppId as string;
+        return { appId: appId, env: "local" };
+      } else {
+        if (env === undefined) {
+          const inputs = getSystemInputs();
```

Comment:
`const inputs = getSystemInputs()` is already called at line 113. Could that `inputs` be reused?